pandas/_libs/lib.pyx

```diff
@@ -1324,12 +1330,12 @@ def infer_dtype(value: object, skipna: bool = True) -> str:
         # e.g. categoricals
         dtype = value.dtype
         if not isinstance(dtype, np.dtype):
-            value = _try_infer_map(value.dtype)
-            if value is not None:
-                return value
+            inferred = _try_infer_map(value.dtype)
+            if inferred is not None:
+                return inferred
 
             # its ndarray-like but we can't handle
-            raise ValueError(f"cannot infer type for {type(value)}")
+            values = np.asarray(value)
```

Comment:
do we have tests that get here?

Reply:
The base extension test and the decimal test that I added in this PR get here (those were raising errors before)

Reply:
Note I am not fully sure about this change. Before it raised an error, now it will typically return "mixed" for random python objects. Both don't seem very useful (or in other words, both are an indication that nothing could be inferred). But I found it a bit inconsistent to raise in this case, while if you would pass a list of the same objects, we would actually infer

Reply:
we are *already* doing this below.  This does a full conversion of the object, I think this will simply kill performance in some cases making this routine very unpredictable. would rather not make this change here.

Reply:
random python objects will be marked as 'mixed' in any event w/o the performance panelty below.